mne/viz/decoding.py

```diff
@@ -129,16 +147,49 @@ def plot_gat_diagonal(gat, title=None, xmin=None, xmax=None, ymin=0., ymax=1.,
     import matplotlib.pyplot as plt
     if ax is None:
         fig, ax = plt.subplots(1, 1)
-    # detect whether gat is a full matrix or just its diagonal
+
+    # Detect whether gat is a full matrix or just its diagonal
     if np.all(np.unique([len(t) for t in gat.test_times_['times_']]) == 1):
         scores = gat.scores_
+    elif train_time == 'diagonal':
+        # Get scores from identical training and testing times even if GAT
+        # is not square.
+        scores = np.zeros(len(gat.scores_))
+        for i, train_time in enumerate(gat.train_times['times_']):
+            for test_times in gat.test_times_['times_']:
+                # find closest testing time from train_time
+                j = np.abs(test_times - train_time).argmin()
+                # check that not more than 1 classifier away
+                if train_time - test_times[j] <= gat.train_times['step']:
+                    scores[i] = gat.scores_[i][j]
```

Comment:
I see lots of single letter variable names. let's avoid this. They're uninformative and tend to intefer with the debugger shortcuts.